platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -954,6 +962,8 @@ private Object parseValue(ComboboxHolder holder, String methodParameterType, Fie
         } else if (null != holder && holder.isEnumCollection()) {
             String genericType = holder.getEnumName();
             parsedValue = TypeHelper.parse(valueAsString, holder.getTypeClassName(), genericType, classLoader);
+        } else if (parsedValue instanceof UUID) {
```

Comment:
@kmadej It should be parsed otherwise we would get an error while invoking the setUuid method (because of the fact that string value cannot be set to a uuid field)